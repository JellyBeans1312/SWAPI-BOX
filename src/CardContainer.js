import React from 'react';
import Card from './Card'

const CardContainer = ({ peopleData }) => {
  const person = peopleData.map(person => {
    return <Card name={person.name} species={person.species} />
  })
  return (
    person
  )
}

export default CardContainer;