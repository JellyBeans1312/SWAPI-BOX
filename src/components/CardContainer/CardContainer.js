import React from 'react';
import Card from '../Card/Card'
import './CardContainer.scss'

const CardContainer = ({ data }) => {
  const card = data.map(person => {
    return <Card name={person.name} species={person.species} />
  })
  return (
    <section className='card-components'>
      {card}
    </section>
  )
}

export default CardContainer;