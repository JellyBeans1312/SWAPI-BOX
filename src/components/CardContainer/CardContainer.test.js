import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
  let wrapper;
  let mockData = [
    'somewhere',
    'over',
    'the',
    'rainbow',
    'stars',
    'something',
    'words'
  ]
  beforeEach(() => {
    wrapper = shallow(
      <CardContainer
        data={mockData}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot() 
  }); 
});
