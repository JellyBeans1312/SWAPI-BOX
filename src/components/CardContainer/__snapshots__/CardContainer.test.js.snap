// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot 1`] = `
<section
  className="card-components"
>
  <Card
    datum="somewhere"
    key="e"
  />
  <Card
    datum="over"
  />
  <Card
    datum="the"
  />
  <Card
    datum="rainbow"
    key="w"
  />
  <Card
    datum="stars"
  />
  <Card
    datum="something"
    key="i"
  />
  <Card
    datum="words"
  />
</section>
`;
